From f0121ed827cf72208f6b1482a3a57bc3537aac93 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 7 Mar 2019 04:13:27 -0800
Subject: [PATCH 0553/1851] v4l: xilinx-demosaic: Add support for 10,12,16 bpc
 RAW and RGB bus formats

commit 4333003e6970752170e3c3b32a98706b13dc0266 from
https://github.com/Xilinx/linux-xlnx.git

Adds support for the different 10, 12, 16 bit per component RAW and
RGBbus formats.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-demosaic.c | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-demosaic.c b/drivers/media/platform/xilinx/xilinx-demosaic.c
index 531e6ec5dbf7..221366ae0ecb 100644
--- a/drivers/media/platform/xilinx/xilinx-demosaic.c
+++ b/drivers/media/platform/xilinx/xilinx-demosaic.c
@@ -146,15 +146,27 @@ xdmsc_is_format_bayer(struct xdmsc_dev *xdmsc, u32 code)
 {
 	switch (code) {
 	case MEDIA_BUS_FMT_SRGGB8_1X8:
+	case MEDIA_BUS_FMT_SRGGB10_1X10:
+	case MEDIA_BUS_FMT_SRGGB12_1X12:
+	case MEDIA_BUS_FMT_SRGGB16_1X16:
 		xdmsc->bayer_fmt = XDEMOSAIC_RGGB;
 		break;
 	case MEDIA_BUS_FMT_SGRBG8_1X8:
+	case MEDIA_BUS_FMT_SGRBG10_1X10:
+	case MEDIA_BUS_FMT_SGRBG12_1X12:
+	case MEDIA_BUS_FMT_SGRBG16_1X16:
 		xdmsc->bayer_fmt = XDEMOSAIC_GRBG;
 		break;
 	case MEDIA_BUS_FMT_SGBRG8_1X8:
+	case MEDIA_BUS_FMT_SGBRG10_1X10:
+	case MEDIA_BUS_FMT_SGBRG12_1X12:
+	case MEDIA_BUS_FMT_SGBRG16_1X16:
 		xdmsc->bayer_fmt = XDEMOSAIC_GBRG;
 		break;
 	case MEDIA_BUS_FMT_SBGGR8_1X8:
+	case MEDIA_BUS_FMT_SBGGR10_1X10:
+	case MEDIA_BUS_FMT_SBGGR12_1X12:
+	case MEDIA_BUS_FMT_SBGGR16_1X16:
 		xdmsc->bayer_fmt = XDEMOSAIC_BGGR;
 		break;
 	default:
@@ -180,7 +192,10 @@ static int xdmsc_set_format(struct v4l2_subdev *subdev,
 				   XDEMOSAIC_MIN_HEIGHT, xdmsc->max_height);
 
 	if (fmt->pad == XVIP_PAD_SOURCE) {
-		if (__format->code != MEDIA_BUS_FMT_RBG888_1X24) {
+		if (__format->code != MEDIA_BUS_FMT_RBG888_1X24 &&
+		    __format->code != MEDIA_BUS_FMT_RBG101010_1X30 &&
+		    __format->code != MEDIA_BUS_FMT_RBG121212_1X36 &&
+		    __format->code != MEDIA_BUS_FMT_RBG161616_1X48) {
 			dev_dbg(xdmsc->xvip.dev,
 				"%s : Unsupported source media bus code format",
 				__func__);
-- 
2.31.1

